whylog/tests/tests_client/__init__.py

```diff
@@ -35,13 +35,91 @@ def test_one(self, test_name):
         output_path = path_join(path, 'expected_output.txt')
 
         whylog_base = WhylogBase(parser=parsers_path, rules=rules_path)
-        whylog_client = WhylogClient(base=whylog_base, open_path=path)
+        # whylog_client = WhylogClient(rulesbase=whylog_base, open_path=path)
 
         with open(input_path, 'r') as f:
             vim_line = f.read()
-        result = whylog_client.get_cause(vim_line)
+        # TODO call get_cause with sens...
+        # result = whylog_client.get_cause(1, vim_line)
 
         with open(output_path, 'r') as f:
             pass
             # remove this comment after implement whylog client and base
             # assert result == f.read()
+
+class TestBacktrackSearcher(TestCase):
+
+    def test_imports_correctness(self):
+        FOO_BAR = "/foo/bar"
+        client = WhylogClient(rulesbase=WhylogBase())
+        bts = searchers.BacktrackSearcher(file_path=FOO_BAR)
+        assert bts._file_path == FOO_BAR
+
+    def _get_log_file_path(self):
+        client_tests_path = "/".join(path_test_files)
+        return "%s/a_few_lines.log" % client_tests_path
+
+    def _get_sample_offset(self, line_num):
+        """
+        Returns the correct offset of line 'line_num'
+        referring to the file a_few_lines.log construction.
+        Be careful when modifying a_few_lines.log.
+        """
+        line_length = 10
+        return line_num * line_length
+
+    def _reverse_from_offset_wrapper(self, backtracker, offset, buf_size=None):
+        if buf_size == None:
+            return [line for line in backtracker._reverse_from_offset(offset)]
+        return [line for line in backtracker._reverse_from_offset(offset, buf_size)]
```

Comment:
clone which can be refactored, for example by using module constant as a default value in both backtracker and here